sdk/python/tests/compiler/compiler_tests.py

```diff
@@ -213,8 +213,16 @@ def test_basic_workflow_without_decorator(self):
       with open(os.path.join(test_data_dir, 'basic_no_decorator.yaml'), 'r') as f:
         golden = yaml.safe_load(f)
 
+      name_to_template = {template['name']: template for template in compiled_workflow['spec']['templates']}
+      for k, v in name_to_template.items():
+        if k in ['exiting', 'get-frequent', 'save']:
```

Comment:
The assertion is true for all pipelines right?

Reply:
What do you mean by "all pipelines"? 'exiting', 'get-frequent', 'save' are just the name of the templates in this pipeline.
But all pipelines should have the labels like sdk version or sdk type. I think it's fine that we just verify labels are added in this pipeline and ignore the metadata field in other UTs. Otherwise the UTs will become quite cumbersome.